sw/host/opentitanlib/src/io/pin_mapper.rs

```diff
@@ -0,0 +1,106 @@
+}
+
+impl PinMapper {
+    /// Given an pin/uart/spi/i2c port name, if the name is a known alias, return the underlying
+    /// name/number, otherwise return the string as is.
+    pub fn get_pin_alias(&self, name: &str) -> Result<String> {
+        Ok(Self::map_name(&self.pin_map, name))
+    }
+
+    pub fn get_pin_config(&self, name: &str) -> Result<&gpio::PinConfiguration> {
+        let name = self.get_pin_alias(name).unwrap();
+        Ok(self
+            .pin_conf_map
+            .get(&name)
+            .ok_or(GpioError::InvalidPinName(name.to_string()))?)
+    }
+
+    pub fn get_pin_level(&self, name: &str) -> Result<bool> {
+        let name = self.get_pin_alias(name).unwrap();
+        Ok(self.get_pin_config(&name)?.level.unwrap_or(false))
+    }
+
+    pub fn get_pin_mode(&self, name: &str) -> Result<gpio::PinMode> {
+        let name = self.get_pin_alias(name).unwrap();
+        Ok(self
+            .get_pin_config(&name)?
+            .mode
+            .unwrap_or(gpio::PinMode::PushPull))
+    }
+
+    pub fn list_gpio_pins(&self) -> Result<Vec<&String>> {
+        Ok(self.pin_map.keys().collect())
+    }
+
+    pub fn list_gpio_pin_alias(&self) -> Result<Vec<&String>> {
+        Ok(self.pin_map.values().collect())
+    }
+
+    pub fn get_spi_config(&self, name: &str) -> Result<&spi::SpiConfiguration> {
+        self.spi_conf_map.get(name).ok_or(
+            TransportError::InvalidInstance(TransportInterfaceType::Spi, name.to_string()).into(),
+        )
+    }
+
+    pub fn get_spi_name(&self, name: &str) -> Result<&String> {
+        Ok(&self.get_spi_config(name)?.underlying_instance)
+    }
+
+    pub fn get_i2c_name(&self, name: &str) -> Result<&String> {
+        self.i2c_map.get(name).ok_or(
+            TransportError::InvalidInstance(TransportInterfaceType::I2c, name.to_string()).into(),
+        )
+    }
+
+    pub fn get_uart_name(&self, name: &str) -> Result<&String> {
+        self.uart_map.get(name).ok_or(
+            TransportError::InvalidInstance(TransportInterfaceType::Uart, name.to_string()).into(),
+        )
+    }
+
+    pub fn get_strapping_conf(
+        &self,
+        name: &str,
+    ) -> Result<&HashMap<String, gpio::PinConfiguration>> {
+        Ok(self
+            .strapping_conf_map
+            .get(name)
+            .ok_or(GpioError::InvalidPinName(name.to_string()))?)
+    }
+
+    /// Given an pin/uart/spi/i2c port name, if the name is a known alias, return the underlying
+    /// name/number, otherwise return the string as is.
+    pub fn map_name(map: &HashMap<String, String>, name: &str) -> String {
+        let name = name.to_uppercase();
+        match map.get(&name) {
+            Some(v) => {
+                if v.eq(&name) {
+                    name
+                } else {
+                    Self::map_name(map, v)
+                }
+            }
+            None => name,
+        }
+    }
```

Comment:
Can we move this out of `PinMapper` and rename it to something like `AliasMapper::map` since it's used for mapping things other than pins in `crate::app`?